### PR TITLE
[2/2] Rename awsiamauth structs and reduce package API surface area 

### PR DESCRIPTION
**Pull request by chrisdoherty4**
### Description

Dependencies: #3476 

Rename the AWS IAM structs to be representative of their function and unexport APIs that aren't required externally any longer. Additionally remove stutter from exported names resulting in `Installer` and `TemplateBuilder` names.

The newly unexported APIs are the result of moving AWS IAM Auth installation from the cluster manager into the awsiamauth package in https://github.com/aws/eks-anywhere/pull/3476. Additionally, I removed a superfluous initializing constructor function for the template builder that is used externally by the EKS-A controller.

### Testing

As in #3476 an AWS IAM Auth E2E test was run locally using the Docker provider.

### CODE REVIEW

File: pkg/awsiamauth/installer.go [hunk 1]
```diff
@@ -40,83 +37,73 @@ var awsIamAuthCaSecretTemplate string
...
-}
-
-func (a *AwsIamAuthTemplateBuilder) GenerateManifest(clusterSpec *cluster.Spec, clusterId uuid.UUID) ([]byte, error) {
-	var clusterIdValue string
-	// If clusterId is Nil; set value as empty string.
-	if clusterId == uuid.Nil {
-		clusterIdValue = ""
-	} else {
-		clusterIdValue = clusterId.String()
+// CreateAndInstallAWSIAMAuthCASecret creates a Kubernetes Secret in cluster containing a
```
Comment (chrisdoherty4) [inline pkg/awsiamauth/installer.go:63]: This is reshuffling function ordering, no logic changes.

File: pkg/awsiamauth/template.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"gopkg.in/yaml.v3"
+
+	"github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/templater"
+)
+
+type TemplateBuilder struct{}
+
+func (a *TemplateBuilder) GenerateManifest(clusterSpec *cluster.Spec, clusterId uuid.UUID) ([]byte, error) {
```
Comment (chrisdoherty4) [inline pkg/awsiamauth/template.go:17]: This is copied from the original `awsiamauth.go`

Comment (g-gaston) [inline pkg/awsiamauth/template.go:17]:
```suggestion
func (t *TemplateBuilder) GenerateManifest(clusterSpec *cluster.Spec, clusterId uuid.UUID) ([]byte, error) {
```

**Review by g-gaston — APPROVED** [review]

**Pull request merged**